doc/website/blog/2023-02-23-simple-cms-with-ent.mdx

```diff
@@ -0,0 +1,777 @@
+
+Inside the Go-template, whatever we pass to it as data is available as "`.`", this explains this line, where we use `range` to iterate over each post:
+```
+{{- range . }}
+```
+Next, we print the title, creation time and the author name, via the `Author` edge:
+```
+<h2>{{ .Title }}</h2>
+<p>
+	{{ .CreatedAt.Format "2006-01-02" }} by {{ .Edges.Author.Name }}
+</p>
+```
+Finally, we print the post body, and close the loop.
+```
+	<p>
+		{{ .Body }}
+	</p>
+{{- end }}
+```
+
+After defining the template, we need to make it available to our program, we embed this template into our binary using the `embed` package ([docs](https://pkg.go.dev/embed)):
+
+```go
+var (  
+   //go:embed templates/*   
+   resources embed.FS  
+   tmpl      = template.Must(template.ParseFS(resources, "templates/*"))  
+)
+```
+
+#### Server code
+
+We continue by defining a type named `server` and a constructor for it, `newServer`. This struct will have receiver methods for each HTTP handler we create and binds the Ent client we created at init to the server code.
+```go
+type server struct {
+	client *ent.Client
+}
+
+func newServer(client *ent.Client) *server {
+	return &server{client: client}
+}
+
+```
+
+Next, let's define the handler for our blog home page. This page should contain a list of all available blog posts:
+
+```go
+// index serves the blog home page
+func (s *server) index(w http.ResponseWriter, r *http.Request) {
+	posts, err := s.client.Post.
+		Query().
+		WithAuthor().
+		All(r.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if err := tmpl.Execute(w, posts); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+```
+
+Let's zoom in on the Ent code here that is used to retrieve the posts from the database:
+```go
+// s.client.Post contains methods for interacting with Post entities
+s.client.Post.
+	// Begin a query.
+	Query().
+	// Retrieve the entities using the `Author` edge. (a `User` instance)
+	WithAuthor().
+	// Run the query against the database using the request context.
+	All(r.Context())
+```
+
+#### The router
+
+To manage the routes for our application, let's use `go-chi` a popular routing library for Go.
+
+```
+go get -u github.com/go-chi/chi/v5
+```
+
+We define the `newRouter` function that sets up our router:
+
+```go
+// newRouter creates a new router with the blog handlers mounted.
+func newRouter(srv *server) chi.Router {
+	r := chi.NewRouter()
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+	r.Get("/", srv.index)
+	return r
+}
+```
+
+In this function, we first instantiate a new `chi.Router`, then register two middlewares:
+* `middleware.Logger` - is a basic access logger that prints out some information on every request our server handles.
+* `middleware.Recoverer` recovers from if our handlers panic, preventing a cause where our entire server will crash because of an application error.
```

Comment:
```suggestion
* `middleware.Recoverer` recovers from when our handlers panic, preventing a case where our entire server will crash because of an application error.
```